packages/hash/frontend/src/components/hooks/use-get-workspace-shortname-by-entity-uuid.ts

```diff
@@ -0,0 +1,75 @@
+import { extractBaseUri } from "@blockprotocol/type-system";
+import { systemUserShortname } from "@hashintel/hash-shared/environment";
+import { types } from "@hashintel/hash-shared/ontology-types";
+import { entityIdFromOwnedByIdAndEntityUuid } from "@hashintel/hash-subgraph";
+import { getRootsAsEntities } from "@hashintel/hash-subgraph/src/stdlib/element/entity";
+import { useEffect, useMemo, useState } from "react";
+import { useBlockProtocolGetEntity } from "./blockProtocolFunctions/knowledge/useBlockProtocolGetEntity";
+import { useGetWorkspaceByShortname } from "./use-get-workspace-by-shortname";
+
+export const useGetWorkspaceShortnameByEntityUuid = (params: {
+  entityUuid: string;
+}): { workspaceShortname?: string; loading: boolean } => {
+  const { entityUuid } = params;
+
+  const [workspaceShortname, setWorkspaceShortname] = useState<string>();
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const { getEntity } = useBlockProtocolGetEntity();
+
+  const {
+    workspace: systemUserWorkspace,
+    loading: loadingSystemUserWorkspace,
+  } = useGetWorkspaceByShortname(systemUserShortname);
+
+  const systemUserOwnedById = useMemo(
+    () =>
+      systemUserWorkspace
+        ? systemUserWorkspace.kind === "user"
+          ? systemUserWorkspace.userAccountId
+          : systemUserWorkspace.orgAccountId
+        : undefined,
+    [systemUserWorkspace],
+  );
+
+  useEffect(() => {
+    if (systemUserOwnedById) {
+      void (async () => {
+        setLoading(true);
+
+        const entityId = entityIdFromOwnedByIdAndEntityUuid(
+          systemUserOwnedById,
```

Comment:
Quick question regarding the logic in this hook, we are doing this because we need to construct the `EntityId` of the user to use in the BP hook. That hook does not allow specifying `EntityUuid` to fetch the user's workspace, right?

Reply:
We are doing this because we need to construct the the `entityId` of the workspace to get it using `getEntity`, in cases where we only have the workspaces `entityUuid`. Therefore we need the `ownedById`, which in the case of workspaces is the "system user"'s `accountId` (because users and orgs are owned by the system user).